oaknational/Oak-Web-Application
# LESQ-977: Redirect user through onboarding on signup

## Description

Music year: [1999](https://www.youtube.com/watch?v=9Ht5RZpzPqw)

- Redirects users to the onboarding page immediately after sign-up (or login if they have not previously completed onboarding)
- Redirects signed-in users to the onboarding page from any other page if they have not previously completed onboarding

## How to test

1. You'll need the use-auth-owa [feature flag enabled](https://eu.posthog.com/project/124/feature_flags/27071)
2. Go to https://deploy-preview-2681--oak-web-application.netlify.thenational.academy/teachers/programmes/art-secondary-ks3-l/units/pop-art-modern-popular-culture-4928/lessons/chila-kumari-burman-part-1-cdh3jc
3. Click on "Download resources"
4. You should see a login screen
5. Sign-up or login
6. You should see the onboarding page
7. Complete the onboarding flow
8. Click continue
3. You should see the download page and be able to download files

## Screenshots


https://github.com/user-attachments/assets/0caedc6b-b6c5-4469-8230-3ce81f116920



**Basic error state if the request to onboard the user fails (server or network error)**
I think this treatment can be improved but for now at least it gives some feedback instead of failing silently.
<img width="604" alt="Screenshot 2024-08-09 at 10 09 47" src="https://github.com/user-attachments/assets/a6401dbe-d27c-4fb7-974c-014e75cdd46f">

### For testing
You can trigger the onboarding flow for an existing user by:

1. Signing out if you're signed 
2. Finding the user in Clerk under the development app and removing `owa:onboarded` from public metadata

<img width="1199" alt="Screenshot 2024-08-19 at 09 44 46" src="https://github.com/user-attachments/assets/4b4de5e3-defe-4c9d-9c42-c4802e2d8977">


  

## Checklist
